cirq-core/cirq/protocols/apply_unitary_protocol.py

```diff
@@ -87,11 +96,26 @@ def __init__(
                 dtype as the target tensor.
             axes: Which axes the unitary effect is being applied to (e.g. the
                 qubits that the gate is operating on).
-
+            subspaces: Which subspace the unitary effect is being applied to on
+                each axis. By default it applies to subspace 0..d-1 on each
+                axis, where d is the dimension of the unitary effect on that
+                axis. Subspaces must be representable as a slice, so the
+                dimensions specified here need to have a consistent step size.
```

Comment:
By slice I mean a (start, end, step). So more precisely, the subspace at each axis needs to have consistent step size. What you specified is fine. But you can't use subspace `[(0, 1, 3)]` on a 4D qudit because that's inconsistent step size within that axis. (This limitation is only due to the fact I couldn't figure out a way to make numpy do the computation on non-slices efficiently. But it may be possible).

I'm kinda 50/50 on whether that limitation makes this more-confusing-than-useful. On one side it works for *any* 2D subspace, which is the most likely use case, and it's a low-maintenance option to have around for if anyone comes across a need for it. On the other side nobody has asked for this in the 2+ years the parent issue has been open.